staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -88,6 +88,38 @@ func ValidateCustomResourceDefinitionSpec(spec *apiextensions.CustomResourceDefi
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("scope"), spec.Scope, []string{string(apiextensions.ClusterScoped), string(apiextensions.NamespaceScoped)}))
 	}
 
+	if len(spec.Versions) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("versions"), ""))
+	} else {
+		storageFlagCount := 0
+		servedFlagCount := 0
+		servedVersions := map[string]bool{}
+		for _, version := range spec.Versions {
+			if version.Storage {
+				storageFlagCount++
+			}
+			if version.Served {
+				servedFlagCount++
+			}
+			if version.Served {
+				if servedVersions[version.Name] {
```

Comment:
this means that non-served versions can be duplicated?

Reply:
I though it is a valid usecase that I have two "v1alpha1"'s for example that I am testing and I can enable either not both. But I don't feel strong about it.

Reply:
IMO such a use-case is confusing. Would check uniqueness in all cases.

Reply:
done